link.go

```diff
@@ -111,6 +122,13 @@ func (link *ToxicLink) Start(name string, source io.Reader, dest io.WriteCloser)
 				"err":   err,
 			}).Warn("Destination terminated")
 		}
+
+		metrics.SentBytesTotal.WithLabelValues(
+			streamDirection,
+			proxyName,
+			listenAddr,
+			upstreamAddr,
+		).Add(float64(bytes))
```

Comment:
It would nice to have a context object with labels values. and then add metrics only in the code. it would remove duplication also partially related in the design to avoid using of global variables.

Reply:
I'm not sure that this is possible in the Prometheus library (if you know how, please let me know). From how I've used these libraries, it's always `<metric> . <apply labels> . <apply values>` and I don't know that it's possible to do something like `<init labels> . <apply metric> . <apply values>`